ydb/docs/en/core/dev/example-app/java/index.md

```diff
@@ -204,10 +173,12 @@ private void selectWithParams(long seriesID, long seasonID) {
 }
 ```
 
-{% include [scan_query.md](../_includes/steps/08_scan_query.md) %}
+{% include [async_requests.md](../_includes/steps/11_async_requests.md) %}
+
+If the expected count of the response rows is large, asynchronous reading is a more preferable way to process them. In this case the `SessionRetryContext` is still used for the retries, because the response parts processing can be interrupted in any moment and after that the entire process of executing has to restart.
```

Comment:
```suggestion
If the expected row count in the response is large, asynchronous reading is the preferred way to process them. In this case, the `SessionRetryContext` is still used for retries because processing response parts can be interrupted at any moment, requiring the entire execution process to restart.
```